src/app/qgsmaptooltrimextendfeature.cpp

```diff
@@ -229,12 +230,18 @@ void QgsMapToolTrimExtendFeature::canvasReleaseEvent( QgsMapMouseEvent *e )
           filter.setLayer( mVlayer );
           match = mCanvas->snappingUtils()->snapToMap( mMapPoint, &filter );
 
-          match.layer()->beginEditCommand( tr( "Trim/Extend feature" ) );
-          match.layer()->changeGeometry( match.featureId(), mGeom );
-          match.layer()->endEditCommand();
-          match.layer()->triggerRepaint();
+          if ( match.layer() )
+          {
+
+            match.layer()->beginEditCommand( tr( "Trim/Extend feature" ) );
+            match.layer()->changeGeometry( match.featureId(), mGeom );
+            match.layer()->addTopologicalPoints( mIntersection );
```

Comment:
Are we sure that mIntersection is linked to our match? what happen if we match and update intersection, then we move again and break before updating mIntersection like [here](https://github.com/troopa81/QGIS/blob/master/src/app/qgsmaptooltrimextendfeature.cpp#L128) or [here](https://github.com/troopa81/QGIS/blob/master/src/app/qgsmaptooltrimextendfeature.cpp#L119)

Reply:
I think so, since mIsModified is my flag for these cases.

Reply:
Yes, but mIsModified is only updated if mIntersection is updated, so it won't be if we break in either of the 2 `if`, no?

Reply:
Yes, but we don't want to update the topology if the geometry isn't modified.
But, maybe I'm missing something.

Reply:
OK, with our conversation, I forced the flag to false even if it's very difficult to produce this case IMHO. :wink: